### PR TITLE
STUB&RES: fix scoped resolve inside stubbed fn body

### PR DESCRIPTION
**Pull request by vlad20012**
```rust
//- main.rs
    mod foo;
    fn main() {
        foo::S.baz();
    }        //^ foo.rs
//- foo.rs
    pub struct S;

    fn foobar() {
        if true {
            impl S {
                pub fn baz(&self) {}
            }
        } else {
            struct S;
        }
    }
```

Works when `foo.rs` is stubbed

### CODE REVIEW

File: src/test/kotlin/org/rust/lang/core/resolve/RsStubOnlyResolveTest.kt [hunk 1]
```diff
@@ -561,4 +561,24 @@ class RsStubOnlyResolveTest : RsResolveTestBase() {
     //- main/foo.rs
         pub struct Bar;
     """, NameResolutionTestmarks.crateRootModule)
+
+    fun `test scoped resolve inside stubbed function body`() = stubOnlyResolve("""
```
Comment (Undin) [inline src/test/kotlin/org/rust/lang/core/resolve/RsStubOnlyResolveTest.kt]: What does this test case check?
Looks like it already works with the current nightly plugin

Comment (vlad20012) [reply src/test/kotlin/org/rust/lang/core/resolve/RsStubOnlyResolveTest.kt]: I will not work if `foo.rs` is stubbed

Comment (Undin) [reply src/test/kotlin/org/rust/lang/core/resolve/RsStubOnlyResolveTest.kt]: Indeed!

File: src/main/kotlin/org/rust/lang/core/stubs/StubImplementations.kt [hunk 2]
```diff
@@ -1108,6 +1109,26 @@ class RsBinaryOpStub(
     }
 }
 
+class RsBlockStub(
+    parent: StubElement<*>?, elementType: IStubElementType<*, *>
+) : StubBase<RsBinaryOp>(parent, elementType) {
```
[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/stubs/StubImplementations.kt]: Sounds reasonable. Done

**Pull request merged**